File: analyze_sales.py
import json
from collections import Counter

file_path = r'c:\Users\zepeg\OneDrive\Área de Trabalho\Nova pasta\santa-brasa-app\metric-s-939ee-default-rtdb-orders-export.json'

with open(file_path, 'r', encoding='utf-8') as f:
    data = json.load(f)

total_revenue = 0
num_orders = len(data)
item_counts = Counter()
item_revenue = Counter()
payment_counts = Counter()
order_type_counts = Counter()
customer_revenue = Counter()
customer_orders = Counter()
daily_sales = Counter()

for order_id, order in data.items():
    total = order.get('total', 0)
    total_revenue += total
    
    payment = order.get('payment', 'N/A')
    payment_counts[payment] += 1
    
    order_type = order.get('orderType', 'N/A')
    order_type_counts[order_type] += 1
    
    customer_name = order.get('customer', {}).get('name', 'Unknown')
    customer_revenue[customer_name] += total
    customer_orders[customer_name] += 1
    
    timestamp = order.get('timestamp', '')
    if timestamp:
        date = timestamp.split('T')[0]
        daily_sales[date] += total
    
    for item in order.get('items', []):
        name = item.get('name', 'Unknown')
        qty = item.get('quantity', 0)
        price = item.get('price', 0)
        item_counts[name] += qty
        item_revenue[name] += (qty * price)

avg_ticket = total_revenue / num_orders if num_orders > 0 else 0

print(f"--- Resumo de Vendas ---")
print(f"Total de Pedidos: {num_orders}")
print(f"Faturamento Total: R$ {total_revenue:.2f}")
print(f"Ticket Médio: R$ {avg_ticket:.2f}")
print(f"\n--- Itens Mais Vendidos (Qtd) ---")
for item, count in item_counts.most_common(5):
    print(f"{item}: {count}")

print(f"\n--- Faturamento por Item ---")
for item, rev in item_revenue.most_common(5):
    print(f"{item}: R$ {rev:.2f}")

print(f"\n--- Formas de Pagamento ---")
for pay, count in payment_counts.items():
    print(f"{pay}: {count}")

print(f"\n--- Tipos de Pedido ---")
for ot, count in order_type_counts.items():
    print(f"{ot}: {count}")

print(f"\n--- Melhores Clientes (Faturamento) ---")
for cust, rev in customer_revenue.most_common(3):
    print(f"{cust}: R$ {rev:.2f} ({customer_orders[cust]} pedidos)")

print(f"\n--- Vendas Diárias ---")
for date, rev in sorted(daily_sales.items()):
    print(f"{date}: R$ {rev:.2f}")

File: analyze_sales.js
const fs = require('fs');

const filePath = 'metric-s-939ee-default-rtdb-orders-export.json';

try {
    const data = JSON.parse(fs.readFileSync(filePath, 'utf8'));
    
    let totalRevenue = 0;
    const numOrders = Object.keys(data).length;
    const itemCounts = {};
    const itemRevenue = {};
    const paymentCounts = {};
    const orderTypeCounts = {};
    const customerRevenue = {};
    const customerOrders = {};
    const dailySales = {};

    for (const orderId in data) {
        const order = data[orderId];
        const total = order.total || 0;
        totalRevenue += total;
        
        const payment = order.payment || 'N/A';
        paymentCounts[payment] = (paymentCounts[payment] || 0) + 1;
        
        const orderType = order.orderType || 'N/A';
        orderTypeCounts[orderType] = (orderTypeCounts[orderType] || 0) + 1;
        
        const customerName = (order.customer && order.customer.name) || 'Unknown';
        customerRevenue[customerName] = (customerRevenue[customerName] || 0) + total;
        customerOrders[customerName] = (customerOrders[customerName] || 0) + 1;
        
        const timestamp = order.timestamp || '';
        if (timestamp) {
            const date = timestamp.split('T')[0];
            dailySales[date] = (dailySales[date] || 0) + total;
        }
        
        (order.items || []).forEach(item => {
            const name = item.name || 'Unknown';
            const qty = item.quantity || 0;
            const price = item.price || 0;
            itemCounts[name] = (itemCounts[name] || 0) + qty;
            itemRevenue[name] = (itemRevenue[name] || 0) + (qty * price);
        });
    }

    const avgTicket = numOrders > 0 ? totalRevenue / numOrders : 0;

    console.log(`--- Resumo de Vendas ---`);
    console.log(`Total de Pedidos: ${numOrders}`);
    console.log(`Faturamento Total: R$ ${totalRevenue.toFixed(2)}`);
    console.log(`Ticket Médio: R$ ${avgTicket.toFixed(2)}`);
    
    console.log(`\n--- Itens Mais Vendidos (Qtd) ---`);
    Object.entries(itemCounts)
        .sort((a, b) => b[1] - a[1])
        .slice(0, 5)
        .forEach(([name, count]) => console.log(`${name}: ${count}`));

    console.log(`\n--- Faturamento por Item ---`);
    Object.entries(itemRevenue)
        .sort((a, b) => b[1] - a[1])
        .slice(0, 5)
        .forEach(([name, rev]) => console.log(`${name}: R$ ${rev.toFixed(2)}`));

    console.log(`\n--- Formas de Pagamento ---`);
    for (const [pay, count] of Object.entries(paymentCounts)) {
        console.log(`${pay}: ${count}`);
    }

    console.log(`\n--- Tipos de Pedido ---`);
    for (const [ot, count] of Object.entries(orderTypeCounts)) {
        console.log(`${ot}: ${count}`);
    }

    console.log(`\n--- Melhores Clientes (Faturamento) ---`);
    Object.entries(customerRevenue)
        .sort((a, b) => b[1] - a[1])
        .slice(0, 3)
        .forEach(([cust, rev]) => {
            console.log(`${cust}: R$ ${rev.toFixed(2)} (${customerOrders[cust]} pedidos)`);
        });

    console.log(`\n--- Vendas Diárias ---`);
    Object.keys(dailySales).sort().forEach(date => {
        console.log(`${date}: R$ ${dailySales[date].toFixed(2)}`);
    });

} catch (err) {
    console.error('Error reading or parsing file:', err);
}
